src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -116,9 +136,14 @@ SGMatrix<float64_t> CGaussianKernel::get_parameter_gradient(const TParameter* pa
 
 float64_t CGaussianKernel::compute(int32_t idx_a, int32_t idx_b)
 {
-    float64_t result=distance(idx_a, idx_b);
-	return std::exp(-result);
+	float64_t result;
+	#pragma omp critical
```

Comment:
@karlnapf the issue now is that this is not thread safe anymore

Reply:
because i assign the value of m_eigen_log_width in kernel_function. Basically would have to do the assignment outside the function, but if a user does ->put("log_width", ..) there is no way of tracking this.. hence the gradient dependent parameters would have to be added to the parameter framework somehow. Or AutoDiffScalar would have a reference instead of a value..